src/cursor/cur_backup_incr.c

```diff
@@ -189,6 +195,12 @@ __wt_curbackup_open_incr(WT_SESSION_IMPL *session, const char *uri, WT_CURSOR *o
     other_cb = (WT_CURSOR_BACKUP *)other;
     cursor->key_format = WT_UNCHECKED_STRING(qqq);
     cursor->value_format = "";
+    /*
+     * Incremental cursors use file cursors, but in a non-standard way. Turn off cursor caching.
+     * Retain the session's original setting and then restore it when we're done.
+     */
+    cb->session_cache_flags = F_ISSET(session, WT_SESSION_CACHE_CURSORS);
+    F_CLR(session, WT_SESSION_CACHE_CURSORS);
```

Comment:
I think you can store the previous `cache_flags` as a local variable and restore it at the end of the function (any time after the open of the cursor).  Turns out that the decision about whether a cursor is cacheable happens when the cursor is opened (in `__curfile_create`), and lives with the cursor in the `WT_CURSTD_CACHEABLE` flag.

Reply:
That's a great simplification. Thanks.